modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/cell_renderers/ActionLinkRenderer.js

```diff
@@ -62,6 +63,11 @@ function ActionLinkRenderer({actions, itemData, itemId, options, value}) {
 					url: formattedHref,
 				});
 			}
+			if (currentAction.target === 'modal-permissions') {
+				event.preventDefault();
+
+				openPermissionsModal(formattedHref);
+			}
```

Comment:
Is it necessary to use `modal-permissions`? I'm thinking we could use `'modal'` target for this, and work towards removing `modal-permissions` from DXP. Looking at [`openPermissionsModal` util](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/utils/modals/index.js#L30-L35), there is nothing special about it, we are using `openModal` under the hood. I believe we just need to set `title` and maybe `size` in `FDSEntries.tsx`, and we can use `modal` target.

Reply:
Good point, I'll use the `modal` target instead and remove this `modal-permissions` type.

Reply:
One major difference I found was that `openPermissionsModal` uses `{openModal} from 'frontend-js-web'` whereas in this file, there is a different `openModal` defined in `frontend-data-set-web`.
- [frontend-data-set-web Modal component](https://github.com/liferay-frontend/liferay-portal/blob/2c5ef593c900e500cbe53e616e17dd884506fc30/modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/modal/Modal.js)
- [frontend-js-web Modal component (iframe portion)](https://github.com/liferay-frontend/liferay-portal/blob/2c5ef593c900e500cbe53e616e17dd884506fc30/modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/modal/Modal.js#L632-L734)

Using `target: 'modal'` as is, ended up with a `lg` sized modal that got stuck on the loading spinner (even though the iframe was loaded underneath).
<img width="1069" alt="Screenshot 2023-08-23 at 11 01 02 AM" src="https://github.com/liferay-frontend/liferay-portal/assets/4496722/40985c0a-87eb-4534-86cf-d5d5e0e8b3c1">

I made 2 additions:

1. I enhanced the `target: 'modal'` to accept `size` and `title`, which was implemented but the values weren't being properly passed in. This is how the object being passed into `itemActions` would look:
```javascript
	{
		data: {
			permissionKey: 'permissions',
			size: 'full-screen',
			title: Liferay.Language.get('permissions'),
		},
		href: permissionsURL,
		icon: 'password-policies',
		label: Liferay.Language.get('permissions'),
		target: 'modal',
	}
```

2. Currently an event `is-loading-modal` must be fired to set loading to false. [(See code)](https://github.com/liferay-frontend/liferay-portal/blob/2c5ef593c900e500cbe53e616e17dd884506fc30/modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/modal/Modal.js#L100-L118) I added an event listener to the iframe element to set `loading` to `false` when the iframe finished loading.

I only found 2 uses of `target: 'modal'` in portal, but this event listener seemed better than expecting somewhere to fire an event to set the modal to false:
- [change-tracking-web](https://github.com/liferay-frontend/liferay-portal/blob/2c5ef593c900e500cbe53e616e17dd884506fc30/modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/display/context/PublicationsDisplayContext.java#L452-L466): To set this to false they call this [here](https://github.com/liferay-frontend/liferay-portal/blob/2c5ef593c900e500cbe53e616e17dd884506fc30/modules/apps/change-tracking/change-tracking-web/src/main/resources/META-INF/resources/publications/manage_collaborators.jsp#L48) 
```javascript
window.top.Liferay.fire('is-loading-modal', {isLoading: false});
```
- [commerce-product-definitions-web](https://github.com/liferay-frontend/liferay-portal/blob/master/modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/display/context/CPDefinitionsDisplayContext.java#L376-L391) (didn't look into the specifics of this one, but they do fire this event in a few places)

Reply:
Oh man. Ideally what we want to do it to migrate FDS Modal to `openModal` from `frontend-js-web`. But that goes beyond the scope of this task. I'm fine with pushing whatever fixes are needed in FDS Modal, but long term we definitely don't want to support both versions.

Reply:
Migration task: [Migrate FDS modals into standard DXP modals](https://liferay.atlassian.net/browse/LPS-194393)